src/reference_tree.ml

```diff
@@ -0,0 +1,237 @@
+                        Xml.Element ("description", _, [Xml.PCData descr])]) ->
+            let vhs = d.value_help in
+            let vhs' = (fmt, descr) :: vhs in
+            {d with value_help=vhs'}
+        | Xml.Element ("multi", _, _) -> {d with multi=true}
+        | Xml.Element ("valueless", _, _) -> {d with valueless=true}
+        | Xml.Element ("constraintErrorMessage", _, [Xml.PCData s]) ->
+            {d with constraint_error_message=s}
+        | Xml.Element ("constraint", _, _) -> load_constraint_from_xml d x
+        | Xml.Element ("hidden", _, _) -> {d with hidden=true}
+        | Xml.Element ("secret", _, _) -> {d with secret=true}
+        | Xml.Element ("keepChildOrder", _, _) -> {d with keep_order=true}
+        | _ -> raise (Bad_interface_definition "Malformed property tag")
+    in Xml.fold aux d x
+
+let rec insert_from_xml basepath reftree xml =
+    match xml with
+    | Xml.Element (tag, _,  _) ->
+        let props = Util.find_xml_child "properties" xml in
+        let data =
+            (match props with
+            | None -> default_data
+            | Some p -> data_from_xml default_data p)
+        in
+        let node_type = node_type_of_string (Xml.tag xml) in
+        let node_owner = try let o = Xml.attrib xml "owner" in Some o
+                         with _ -> None
+        in
+        let data = {data with node_type=node_type; owner=node_owner} in
+        let name = Xml.attrib xml "name" in
+        let path = basepath @ [name] in
+        let new_tree = Vytree.insert reftree path data in
+        (match node_type with
+        | Leaf -> new_tree
+        | _ ->
+            let children = Util.find_xml_child "children" xml in
+            (match children with
+             | None -> raise (Bad_interface_definition (Printf.sprintf "Node %s has no children" name))
+             | Some c ->  List.fold_left (insert_from_xml path) new_tree (Xml.children c)))
+    | _ -> raise (Bad_interface_definition "PCData not allowed here")
+
+let load_from_xml reftree file =
+    let xml_to_reftree xml reftree =
+        match xml with
+        | Xml.Element ("interfaceDefinition", attrs, children) ->
+            List.fold_left (insert_from_xml []) reftree children
+        | _ -> raise (Bad_interface_definition "Should start with <interfaceDefinition>")
+    in
+    try
+        let xml = Xml.parse_file file in
+        xml_to_reftree xml reftree
+    with
+    | Xml.File_not_found msg -> raise (Bad_interface_definition msg)
+    | Xml.Error e -> raise (Bad_interface_definition (Xml.error e))
+
+(* Validation function *)
+
+let has_illegal_characters name =
+    (** Checks if string name has illegal characters in it.
+        All whitespace, curly braces, square brackets, and quotes
+        are disallowed due to their special significance to the curly config
+        format parser *)
+    try Some (Pcre.get_substring (Pcre.exec ~pat:"[\\s\\{\\}\\[\\]\"\'#]" name) 0)
```

Comment:
I think we could use the `{| |}` literal syntax here to reduce the number of back slashes. When I wrote that code, that syntax hadn't existed yet but now I'm quite a fan of that syntax, whenever I don't need escape sequences in string literals.

Reply:
Agreed. Note that that part of the code for the value_checker is not needed for this version and was dropped in a later commit.